assets/src/edit-story/components/panels/link/index.js

```diff
@@ -99,15 +101,33 @@ function LinkPanel({ selectedElements, pushUpdateForObject }) {
     if (!isValidUrl(urlWithProtocol)) {
       return;
     }
+    const invalidLinkMessage = __('This is an invalid link!', 'web-stories');
     setFetchingMetadata(true);
     getLinkMetadata(urlWithProtocol)
       .then(({ title, image }) => {
-        updateLinkFromMetadataApi({ url: urlWithProtocol, title, icon: image });
+        if (title && image) {
+          return updateLinkFromMetadataApi({
+            url: urlWithProtocol,
+            title,
+            icon: image,
+          });
+        }
+
+        // This is necessary since the API is returning empty image/title fields sometimes.
+        const invalidLinkError = new Error(invalidLinkMessage);
+        invalidLinkError.code = 'invalid_url';
+        throw invalidLinkError;
       })
       .catch((reason) => {
-        if (reason?.code === 'rest_invalid_url') {
-          return;
+        if (
```

Comment:
Why do we care here what the error code is here? Any error state, should return this message.